Introduce Data type to use with serde

zummenix opened the pull request:
A dedicated `Data` type to use with serde.

Since the `Data` is supposed to be used only for serialization/deserialization purposes with serde, maybe the whole module `data` should be under the `#[cfg(feature = "serde"` flag?

Closes #87

ebarnard commented on src/data.rs:
```diff
@@ -0,0 +1,119 @@
+    }
+}
+
+impl From<Vec<u8>> for Data {
```
Might be worth having `From<Data> for Vec<u8>` as well.

zummenix replied:
I don't know. What is the best practice in rust? If we implement `From<Data> for Vec<u8>` do we still need `into_inner` method?

ebarnard commented:
I think for the moment let's put the whole thing under `#[cfg(feature = "serde")]`.

zummenix commented:
- put the whole `mod data` module under `#[cfg(feature = "serde")]`
- implemented `From<Data> for Vec<u8>`
- removed `into_inner` method
- additionally implemented `AsMut<[u8]> for Data`
- rebased
